huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_environment.go

```diff
@@ -0,0 +1,147 @@
+package apigateway
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/apigw/shared/v1/environments"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceEnvironment() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceEnvironmentCreate,
+		ReadContext:   resourceEnvironmentRead,
+		UpdateContext: resourceEnvironmentUpdate,
+		DeleteContext: resourceEnvironmentDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"create_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceEnvironmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	apigClient, err := cfg.ApiGatewayV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating shared API gateway client (v1) : %s", err)
+	}
+
+	createOpts := environments.CreateOpts{
+		Name:        d.Get("name").(string),
+		Description: d.Get("description").(string),
+	}
+	log.Printf("[DEBUG] Create API gateway shared options: %s", utils.MarshalValue(createOpts))
```

Comment:
```suggestion
	log.Printf("[DEBUG] Create API gateway shared options: %v", createOpts)
```

Reply:
I wrote this like you before, but Xiangdong suggest me use this way to DEBUG. Can both methods be used?

Reply:
> I wrote this like you before, but Xiangdong suggest me use this way to DEBUG. Can both methods be used?

I wrote this like you before, but Xiangdong suggest me use this way to DEBUG. Can both methods be used?

Reply:
> 
I have fixed this problem, thanks for your valuable suggestions